File: app/src/main/java/hanas/aptacy/moodlamp/MainActivityView.java
package hanas.aptacy.moodlamp;

import android.graphics.Bitmap;

import com.spotify.android.appremote.api.ConnectionParams;
import com.spotify.android.appremote.api.Connector;

import java.util.List;

import hanas.aptacy.moodlamp.services.pojo.LeadingColorBody;

public interface MainActivityView {
    void setSongInfo(String title, String artistName, String imageURLStr);
    void setColorsOfNewAlbumCover(List<LeadingColorBody> leadingColors);
    void showSpotifyConnectionError(Throwable throwable);
    void showLeadingColorServiceResponseError(Throwable t);
    void connectWithSpotify(ConnectionParams connectionParams, Connector.ConnectionListener spotifyConnectionListener);
    void setWallpaperFromBitmap(Bitmap bitmap);
    int[] getScreenSize();
}

File: app/src/main/java/hanas/aptacy/moodlamp/MainActivityView.kt
package hanas.aptacy.moodlamp

import android.graphics.Bitmap
import hanas.aptacy.moodlamp.services.leadingcolors.pojo.LeadingColorBody

interface MainActivityView {
    fun setSongInfo(title: String?, artistName: String?, imageURLStr: String?)
    fun setColorsOfNewAlbumCover(leadingColors: List<LeadingColorBody>)
    fun showSpotifyConnectionError(throwable: Throwable?): Unit
    fun showLeadingColorServiceResponseError(t: Throwable?)
    fun setWallpaperToActivityBackground(bitmap: Bitmap?)
    fun setWallpaperFromBitmap(bitmap: Bitmap?)
    val screenSize: IntArray?
    var wallpaperSp: Bitmap?
    fun setFabColor(prominentColor: Int)
}